src/config.ts

```diff
@@ -23,15 +23,59 @@ export interface Config {
   fetch?: (input: string, init?: Req) => Promise<Res>
   arrayMode?: boolean
   fullResult?: boolean
+  decode?: DecodeConfig
 }
 
 export interface ExecuteOptions {
   arrayMode?: boolean
   fullResult?: boolean
+  decode?: DecodeConfig
 }
 
 export interface TxOptions {
   isolation?: 'READ COMMITTED' | 'REPEATABLE READ'
 }
 
 export type ExecuteArgs = object | any[] | null
+
+export const enum ColumnType {
+  TINYINT = 'TINYINT',
+  UNSIGNED_TINYINT = 'UNSIGNED TINYINT',
+  SMALLINT = 'SMALLINT',
+  UNSIGNED_SMALLINT = 'UNSIGNED SMALLINT',
+  MEDIUMINT = 'MEDIUMINT',
+  INT = 'INT',
+  UNSIGNED_INT = 'UNSIGNED INT',
+  YEAR = 'YEAR',
+  FLOAT = 'FLOAT',
+  DOUBLE = 'DOUBLE',
+  BIGINT = 'BIGINT',
+  UNSIGNED_BIGINT = 'UNSIGNED BIGINT',
+  DECIMAL = 'DECIMAL',
+  CHAR = 'CHAR',
+  VARCHAR = 'VARCHAR',
+  BINARY = 'BINARY',
+  VARBINARY = 'VARBINARY',
+  TINYTEXT = 'TINYTEXT',
+  TEXT = 'TEXT',
+  MEDIUMTEXT = 'MEDIUMTEXT',
+  LONGTEXT = 'LONGTEXT',
+  TINYBLOB = 'TINYBLOB',
+  BLOB = 'BLOB',
+  MEDIUMBLOB = 'MEDIUMBLOB',
+  LONGBLOB = 'LONGBLOB',
+  DATE = 'DATE',
+  TIME = 'TIME',
+  DATETIME = 'DATETIME',
+  TIMESTAMP = 'TIMESTAMP',
+  BIT = 'BIT',
+  JSON = 'JSON'
+}
+
+export type DecodeConfig = Partial<
```

Comment:
I prefer `  [key in ColumnType]?: (rawValue: string) => any` rather than Partial